src/blas/backends/syclblas/syclblas_common.hpp

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+* Copyright Codeplay Software
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _SYCLBLAS_COMMON_HPP_
+#define _SYCLBLAS_COMMON_HPP_
+
+#include "sycl_blas.h"
+#include "oneapi/mkl/types.hpp"
+
+#include <tuple>
+#include <utility>
+
+namespace oneapi {
+namespace mkl {
+namespace blas {
+namespace syclblas {
+
+namespace detail {
+using policy_t = ::blas::codeplay_policy;
+using policy_handler_t = ::blas::PolicyHandler<policy_t>;
+
+// SYCL-BLAS executor type. Constructed with sycl::queue.
+using executor_t = ::blas::Executor<detail::policy_handler_t>;
+
+// SYCL-BLAS buffer iterator. Constructed with sycl::buffer<ElemT,1>
+template <typename ElemT>
+using buffer_iterator_t = ::blas::BufferIterator<ElemT, detail::policy_t>;
+
+/** A trait for obtaining equivalent SYCL-BLAS API types from oneMKL API
+ *  types.
+ * 
+ *  @tparam InputT is the oneMKL type.
+ *  syclblas_type<InputT>::type should be the equivalent SYCL-BLAS type.
+**/
+template <typename InputT>
+struct syclblas_type;
+
+#define DEF_SYCLBLAS_TYPE(onemkl_t, syclblas_t) \
+    template <>                                 \
+    struct syclblas_type<onemkl_t> {            \
+        using type = syclblas_t;                \
+    };
+
+DEF_SYCLBLAS_TYPE(sycl::queue, executor_t)
+DEF_SYCLBLAS_TYPE(int64_t, int32_t)
```

Comment:
Converting 64b integers to 32b seems dangerous. If 64b is not supported by SYCL-BLAS, should we not throw?

Reply:
[7474da1](https://github.com/hjabird/oneMKL/pull/1/commits/7474da1e0cb75e5305223ae97cf5611b08007753) now uses `int64_t` instead of `int32_t` for indexing.

This depends on https://github.com/codeplaysoftware/sycl-blas/pull/339.